src/java.base/share/classes/sun/security/util/math/intpoly/IntegerPolynomial.java

```diff
@@ -61,7 +61,14 @@
  *
  */
 
-public abstract class IntegerPolynomial implements IntegerFieldModuloP {
```

Comment:
Although we only have several implementations, I think this class is meant to be freely extendable for whatever new modulus.

Reply:
We can always add new ones to the permits clause later.